core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -146,15 +154,17 @@ public TenantObject setTrustAnchor(final PublicKey publicKey, final X500Principa
         Objects.requireNonNull(publicKey);
         Objects.requireNonNull(subjectDn);
 
-        final JsonObject trustedCa = new JsonObject();
-        trustedCa.put(TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, subjectDn.getName(X500Principal.RFC2253));
-        trustedCa.put(TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, publicKey.getEncoded());
-        return setProperty(TenantConstants.FIELD_PAYLOAD_TRUSTED_CA, trustedCa);
+        final JsonObject trustedCa = new JsonObject()
+                .put(TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, subjectDn.getName(X500Principal.RFC2253))
+                .put(TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, publicKey.getEncoded());
+
+        return addTrustedCaConfiguration(trustedCa);
     }
 
     /**
      * Sets the trusted certificate authority to use for authenticating
-     * devices of this tenant.
+     * devices of this tenant. The certificate must contain a {@code NonNull} and non-empty
```

Comment:
this method should probably be changed to support *adding* a trust anchor instead of setting one, i.e. *addTrustAnchor*